### PR TITLE
Dashrews/ROX-15786 migration timeouts

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

Some Postgres to Postgres migrations may take a significant amount of time.  The timeout periods need to be extended for those long running jobs.  It was discovered during this effort that the index on clusterid is no longer needed as the partitions take care of that.  Simply eliminating that index greatly improved the performance of the migration without impacting queries because it was essentially indexing the entire partition for no reason.  We effectively accomplish the same thing simply be using the partitions.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Migration tests and lots of scale tests of migrations to see how it performed and if it would succeed at all.  This is how the issue with the clusterid index was discovered.

### CODE REVIEW

File: pkg/postgres/pgadmin/admin_utils.go [hunk 1]
```diff
@@ -191,7 +193,9 @@ func AnalyzeDatabase(config *postgres.Config, dbName string) error {
 	// Close the admin connection pool
 	defer connectPool.Close()
 
-	_, err := connectPool.Exec(context.Background(), "ANALYZE")
+	ctx, cancel := context.WithTimeout(context.Background(), analyzeTimeout)
+	defer cancel()
+	_, err := connectPool.Exec(ctx, "ANALYZE")
 
 	log.Debug("Anaylze done")
```
Comment (connorgorman) [inline pkg/postgres/pgadmin/admin_utils.go:200]: Analyze

File: migrator/clone/postgres/db_clone_manager_impl.go [hunk 2]
```diff
@@ -279,6 +279,11 @@ func (d *dbCloneManagerImpl) doPersist(cloneName string, prev string) error {
 		}
 	}
 
+	err = pgadmin.AnalyzeDatabase(d.adminConfig, CurrentClone)
+	if err != nil {
+		log.Warnf("unable to force analyze restore database:  %v", err)
```
Comment (connorgorman) [inline migrator/clone/postgres/db_clone_manager_impl.go:284]: extra space after :

File: migrator/migrations/m_172_to_m_173_network_flows_partition/migration.go [hunk 3]
```diff
@@ -39,37 +38,44 @@ var (
 
 // MigrateToPartitions updates the btree network flow indexes to be hash
 func MigrateToPartitions(gormDB *gorm.DB, db *postgres.DB) error {
+	parentCtx := context.Background()
```
Comment (connorgorman) [inline migrator/migrations/m_172_to_m_173_network_flows_partition/migration.go:41]: just name this ctx?

File: migrator/migrations/m_172_to_m_173_network_flows_partition/migration.go [hunk 4]
```diff
@@ -39,37 +38,44 @@ var (
...
+
+	err := analyzeOldTable(parentCtx, db)
+	if err != nil {
+		log.WriteToStderrf("unable to analyze network_flows.  Will continue processing though it may be slow. %v", err)
+	}
+
 	// First get the distinct clusters in the network_flows table
-	clusters, err := getClusters(db)
+	clusters, err := getClusters(parentCtx, db)
```
Comment (connorgorman) [inline migrator/migrations/m_172_to_m_173_network_flows_partition/migration.go:49]: again curious if we shouldn't just read clusters here. Seems much faster

File: migrator/migrations/m_172_to_m_173_network_flows_partition/migration.go [hunk 5]
```diff
@@ -103,11 +109,49 @@ func MigrateToPartitions(gormDB *gorm.DB, db *postgres.DB) error {
...
+func getPreviousCount(parentCtx context.Context, store previous.FlowStore) (int, error) {
+	// Due to heavy load with network flows, each call to the store could take a long time so we will
+	// pass a context with the migration timeout to them.
+	ctx, cancel := context.WithTimeout(parentCtx, types.DefaultMigrationTimeout)
+	defer cancel()
+
+	return store.Count(ctx)
+}
+
+func getDestinationCount(parentCtx context.Context, store updated.FlowStore) (int, error) {
```
Comment (connorgorman) [inline migrator/migrations/m_172_to_m_173_network_flows_partition/migration.go:121]: its typically okay to overwrite the context so this param can be ctx

**Review by connorgorman — COMMENTED** [review]
added some comments

File: migrator/migrations/m_172_to_m_173_network_flows_partition/migration.go (hunk 4 above)
Comment (dashrews78) [reply migrator/migrations/m_172_to_m_173_network_flows_partition/migration.go:49]: I did.  But pulling in the store and all that nonsense seemed like overkill so I just did `getClustersStmt := "select distinct id from clusters;"` in `getClusters`

**Review by connorgorman — APPROVED** [review]

**Pull request merged**